website/source/docs/agent/options.html.md

```diff
@@ -192,166 +192,180 @@ will exit with an error at startup.
   mitigate node startup race conditions when automating a Consul cluster
   deployment.\
 
-*   <a name="_retry_join"></a><a href="#_retry_join">`-retry-join`</a> - Similar
-    to [`-join`](#_join) but allows retrying a join if the first
-    attempt fails. The list should contain IPv4 addresses with optional Serf
-    LAN port number also specified or bracketed IPv6 addresses with optional
-    port number — for example: `[::1]:8301`. This is useful for cases where we
-    know the address will become available eventually.
+<a name="_retry_join"></a>
 
-    As of Consul 0.9.1 the cloud provider specific discovery of nodes has been
-    moved to the https://github.com/hashicorp/go-discover library which provides
-    a unified query interface for different providers. To use retry join for a
-    supported cloud provider provide a `-retry-join 'provider=xxx key=val key=val
-    ...'` parameter with the provider specific values as described below. This
-    can be combined with static IP addresses and names or even multiple
-    `go-discover` configurations for different providers. This deprecates and
-    replaces the `-retry-join-ec2-*`, `-retry-join-azure-*` and
-    `-retry-join-gce-*` parameters and their usage will be translated to a
-    corresponding `go-discover` config string.
+* `-retry-join` - Similar to [`-join`](#_join) but allows retrying a join if the
+  first attempt fails. This is useful for cases where we know the address will
+  become available eventually. The list should contain IPv4 or IPv6 addresses
+  and optionally the Serf LAN port number.specified or bracketed IPv6 addresses
+  with optional port number. This is useful for cases where we know the address
+  will become available eventually.
 
-    The supported providers for retry join at this point are Amazon EC2,
-    Microsoft Azure, Google Cloud and Softlayer.
-
-    * Amazon EC2
-
-        `provider=aws tag_key=xxx tag_value=xxx [region=xxx] [access_key_id=xxx] [secret_access_key=xxx]`
-
-        This returns the first private IP address of all servers in the given region
-        which have the given `tag_key` and `tag_value`. If the region is omitted it
-        will be discovered through the local instance's [EC2 metadata
-        endpoint](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html).
-
-        Authentication is handled in the following order:
+    ```sh
+    $ consul agent -retry-join "[::1]:8301"
```

Comment:
Can you pull this language up here:

> This can be combined with static IP or DNS addresses or even multiple configurations for different providers.

This is a great chance to make sure people know they can use DNS names here, and that might even be the right thing to put in the example since that's the most useful, something like `consul-servers.mydomain.com` (though it doesn't have to be a server).

It might even be worth a section here on DNS with a mention of Terraform keeping a DNS record up to date for joining.